packages/ceramic-core/src/store/level-state-store.ts

```diff
@@ -0,0 +1,171 @@
+import CID from "cids"
+import Level from 'level-ts';
+
+import type Ipfs from 'ipfs'
+import Document, { AnchorStatus, DocState } from "../document"
+import StateStore from "./state-store"
+
+/**
+ * Level backed State Store
+ */
+export default class LevelStateStore implements StateStore {
+    private store: any
+
+    constructor(private ipfs: Ipfs.Ipfs, private storePath: string) {
+    }
+
+    /**
+     * Open pinning service
+     */
+    async open(): Promise<void> {
+        this.store = new Level(this.storePath);
+    }
+
+    /**
+     * Pin document
+     * @param document - Document instance
+     * @param pinOnIpfs - Pin logs on IPFS
+     */
+    async pin(document: Document, pinOnIpfs = true): Promise<void> {
+        const { state } = document;
+        await this.store.put(document.id, this._serializeState(state))
+
+        if (pinOnIpfs) {
+            const { log } = state;
+            const pinPromises = log.map(async (cid) => {
+                return this.ipfs.pin.add(cid.toString(), {
+                    recursive: false,
+                })
+            })
+            await Promise.all(pinPromises);
+        }
+    }
+
+    /**
+     * Serializes document state
+     * TODO move to common utility
```

Comment:
👍 Let's do this as part of the refactor!

Reply:
yep! 👍